### DISCUSSION TITLE
How to debug AWS errors?

### DISCUSSION THREAD

**Discussion by NeilJed** [vectordotdev/vector#14683] [General]
I'm using the S3 source with SQS and we've noticed we're getting quite a few network errors fetching the files from S3, mostly "TransientError: connection close before message completed". Rate is about 10 per hour at peak

Is there a way to get more verbose debugging of the AWS API calls to try and work out what the issue is? I have a vauge recollection on another thread where there was an environment variable we could set to just get verbose debugging on AWS calls but I can't recall it.

**Comment by jszwedko** [comment]
You could try enabling AWS debugging by setting an environment variable: `VECTOR_LOG=aws_config=debug`. Just a note that Vector will retry those requests.

**Comment by NeilJed** [comment]
Thanks for the clarification.

One question though - when retrying does it raise the error when all of the retries of failed or before a retry is attempted?

**Reply by jszwedko** [reply]
Those messages are warnings rather than errors. Or at least, they should be being logged at the `WARN` level. Let me know if you see otherwise though.
